### PR TITLE
test(coverage): B9 — gateway/connectors remaining 19 files to ≥80% branch (90→71)

### PR DESCRIPTION
**Pull request by asafgolombek**
## Summary

Sub-project **B9** of the True Coverage Program (close per-subsystem branch gaps). Raises the last 19 below-floor `gateway/connectors` files above the ≥80% line+branch floor — the `_lib` helpers, the remaining REST/AWS connectors, the local/index connectors, and the shared connector test-helper. **The connectors subsystem is now fully cleared** in the branch-coverage baseline.

Files cleared (main branch% before):
- **`_lib` helpers:** `pagination` (71.43), `fetch-outcome` (75), `http` (76.92), `rate-limit-observer` (77.78), `gitlab/events` (74.03), `gitlab/pipelines` (76.56)
- **REST / AWS:** `fastmail` (64.36), `cloudwatch` (75.51, AWS `RunAwsCli`), `pagerduty` (76.47), `zoom` (77.88), `iac` (75)
- **local / index:** `connector-catalog` (60 — line+branch), `github-index-repos` (66.67), `user-mcp-store` (72.22), `filesystem-v2-sync` (75.64), `obsidian-daily-note` (77.78), `obsidian-parsing` (79.63), `openapi-indexer-config` (78.38)
- **test helper:** `connector-sync-test-helpers` (62.5)

## Approach

- `_lib` helpers unit-tested directly with crafted inputs; REST connectors via URL-keyed `globalThis.fetch` fakes (the `connector-sync-test-helpers` pattern); `cloudwatch` via an injected `RunAwsCli`; pure parsers with crafted strings.
- **One minimal DI seam:** `filesystem-v2-sync` `gitLogRecords` gains an optional `spawn?: SpawnFn = Bun.spawn` (mirroring the existing `gitBlameLinePorcelain` DI pattern) so the git-log branches are deterministic without real git.
- `iac-sync`'s defensive `lambdaRow?.c ?? 0` arms covered via a delegating-proxy db that returns no row for the lambda `COUNT(*)`.
- **No `any`, no `mock.module`, no `biome-ignore`.** Now-relative date fixtures (no hardcoded dates / `setSystemTime`).

## Baseline

- `coverage-baseline.json` 90 → **71** — 19-file removal.
- `people/person-store.ts` kept at main's 79.21 branch (untouched by this PR; the Docker reseed's ≥80 is environmental over-cover — the CI merge lcov is authoritative and measured 79.21 at B7).

Reseed: Docker `oven/bun:latest` (bun 1.3.14 = CI, Linux-authoritative).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Substantially expanded test coverage across many connectors and helpers (GitLab, GitHub, Fastmail, CloudWatch, PagerDuty, Zoom, Filesystem, Obsidian, OpenAPI, and more), adding edge-case, pagination, error-handling, parsing, and integration-style suites plus in-memory test utilities.

* **Chores**
  * Updated coverage baseline metadata and pruned obsolete coverage entries.
  * Exported an internal helper to improve testability by allowing alternative process-runner injection during tests.

### CODE REVIEW

File: packages/gateway/src/connectors/fastmail-sync.test.ts [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+      return new Response(JSON.stringify(makeSession()), { status: 200 });
+    }) as typeof fetch;
+
+    // Use two-call fake so emails also respond
+    globalThis.fetch = twoCallFetch(makeSession(), makeEmailsResponse([]));
+    const sync = makeSyncable();
+    await sync.sync(syncTestContext(db, tokenVault()), null);
+    // sessionUrl should contain default base URL (api.fastmail.com)
+    void sessionUrl; // set by first fake but overridden; test just checks no crash
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/gateway/src/connectors/fastmail-sync.test.ts:165]: Fixed in 9d6febdb — both base-URL tests previously captured the request URL then overwrote `globalThis.fetch` with `twoCallFetch` (which doesn't capture), so nothing was asserted. They now use a capturing two-call fake and assert the session endpoint: trailing-slash `base_url` → `https://custom.fastmail.example/jmap/session` (no `//`); absent `base_url` → `DEFAULT_BASE_URL` `https://api.fastmail.com/jmap/session`.

File: packages/gateway/src/connectors/fastmail-sync.test.ts [hunk 2]
```diff
@@ -0,0 +1,796 @@
...
+      return new Response(JSON.stringify(makeEmailsResponse([])), { status: 200 });
+    }) as typeof fetch;
+
+    const sync = makeSyncable();
+    await sync.sync(syncTestContext(db, tokenVault("my-secret-token")), null);
+    expect(capturedHeaders.length).toBeGreaterThanOrEqual(2);
+    for (const h of capturedHeaders) {
+      expect(h).toBe("Bearer my-secret-token");
+    }
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/gateway/src/connectors/fastmail-sync.test.ts:708]: Fixed in 9d6febdb — the test now persists an email and asserts the secret token never appears in the returned cursor or in any persisted item row (title/body_preview/metadata/url/canonical_url), in addition to the existing Authorization-header check.

File: packages/gateway/src/connectors/obsidian-daily-note.test.ts [hunk 3]
```diff
@@ -49,3 +49,159 @@ test("resolveDailyNotePath emits a warning when the JSON is malformed", () => {
...
+  mkdirSync(join(root, ".obsidian"), { recursive: true });
+  writeFileSync(join(root, ".obsidian", "daily-notes.json"), "[1,2,3]");
+  // Arrays are typeof "object" but NOT hit by the null check. They pass through
+  // as a Record, so folder/format will be missing — folder defaults to "" and
+  // format defaults to "YYYY-MM-DD". No warning expected.
+  const now = new Date();
+  const out = resolveDailyNotePath(root, now);
+  expect(out.relativePath).toMatch(/^\d{4}-\d{2}-\d{2}\.md$/);
+  expect(out.warning).toBeUndefined();
+});
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/gateway/src/connectors/obsidian-daily-note.test.ts]: Fixed in 9d6febdb — renamed to "does NOT warn for a JSON array (treated as Record → defaults)". A JSON array is `typeof "object"` and non-null, so it passes the object guard, becomes a Record with missing folder/format, and falls through to defaults with no warning — which is exactly what the assertion checks. (The sibling JSON-number test is correctly named: a number is not an object, so it does warn.)
